kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/BigQuerySinkTask.java

```diff
@@ -604,21 +617,43 @@ public void start(Map<String, String> properties) {
     enableRetries = config.getBoolean(BigQuerySinkConfig.ENABLE_RETRIES_CONFIG);
   }
 
-  private void initializeStorageApiBasicMode() {
-    if(testStorageWriteApi != null) {
-      logger.info("Starting task with Test Storage Write API Default Stream...");
+  private void initializeStorageApiMode() {
+    if (testStorageWriteApi != null) {
+      logger.info("Starting task with Test Storage Write API Stream...");
       storageApiWriter = testStorageWriteApi;
+      batchHandler = testStorageApiBatchHandler;
+      loadExecutor = Executors.newScheduledThreadPool(1);
+      loadExecutor.scheduleAtFixedRate(batchHandler::createNewStream, 10, 10, TimeUnit.SECONDS);
```

Comment:
Any reason we are going with value 10 for initial delay and period?

Reply:
This is part of test block. Real block takes user input.

Reply:
Just curious - where is the test block being called, i.e, what cases do we have testStorageWriteApi not null? Also is it not possible to just update the config values to 10 instead of having a separate block since these values are configurable.

Reply:
We have a different constructor for testing [ref](https://github.com/confluentinc/kafka-connect-bigquery/pull/312/files#diff-ad5086c1123b437f762b169166c133be1b56528bd22eba426ec7dea10110f9b2R157) . Only when created via that constructor testStorageWriteApi would be not null. Config range starts from 1 minute. 1-2 minutes felt too long for unit test and 10 seconds if passed via config would have failed.